app/views/hostgroups/index.html.erb

```diff
@@ -16,6 +18,14 @@
           <%= label_with_link(hostgroup, 150, authorizer) %>
         </td>
         <td>
+          <%= link_to hostgroup.hosts_count, hosts_path(:search => "hostgroup_fullname = #{hostgroup}") %>
+        </td>
+        <td>
+          <% if hostgroup.children.any? %>
```

Comment:
This causes N+1 queries on hostgroup page - each hostgroup checks if it has children (`SELECT COUNT(*) FROM "hostgroups" WHERE "hostgroups"."ancestry" = '7'` for example)